pkg/sink/sink.go

```diff
@@ -136,3 +155,64 @@ func createRequestURI(apiVersion, namePlural, namespace string) string {
 	uri = path.Join(uri, namePlural)
 	return uri
 }
+
+func (r Resource) secureEndpoint(trigger triggersv1.Trigger, headers http.Header, payload []byte) error {
+
+	params := []pipelinev1.Param{}
+
+	params = append(params, trigger.TriggerValidate.Params...)
+	params = append(params, pipelinev1.Param{
+		Name: "Payload",
+		Value: pipelinev1.ArrayOrString{
+			Type:      pipelinev1.ParamTypeArray,
+			StringVal: string(payload),
+		},
+	})
+
+	for key := range headers {
+		params = append(params, pipelinev1.Param{
+			Name: key,
+			Value: pipelinev1.ArrayOrString{
+				Type:      pipelinev1.ParamTypeArray,
+				StringVal: headers.Get(key),
+			},
+		})
+	}
+
+	tr := &pipelinev1.TaskRun{
```

Comment:
Testing idea: If this was a separate function that took the event and returned a TaskRun spec, it would be simple to add focused unit tests that test that we generate the TaskRun with the correct spec.

Reply:
Did this.